controller/src/main/java/org/jboss/as/controller/notification/NotificationRegistry.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2016, Red Hat, Inc., and individual contributors as indicated
+ * by the @authors tag.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jboss.as.controller.notification;
+
+import org.jboss.as.controller.PathAddress;
+
+/**
+ * Interface exposed to the users to register notifications via the {@link NotificationRegistrar}
+ *
+ * @author Kabir Khan
+ */
+public interface NotificationRegistry {
```

Comment:
ModelController's 

NotificationHandlerRegistration getNotificationRegistry()

is really poorly named. :(

The method name is reasonable but the return type is kind of yuck.

We could perhaps make NotificationHandlerRegistration a subinterface of NotificationRegistry since they share the one method.

This comment isn't really about your PR here; it's more me not liking something in what we have already and fishing for thoughts about how to make it better.  :)